tests/unit/model_tools/introspection/test_sqlalchemy.py

```diff
@@ -0,0 +1,208 @@
+from typing import Union
+from unittest.mock import ANY
+
+from sqlalchemy import Column, String
+from sqlalchemy.orm import Mapped, declarative_base, mapped_column
+
+from adaptix._internal.model_tools.definitions import (
+    DefaultFactory,
+    DefaultValue,
+    InputField,
+    InputShape,
+    NoDefault,
+    OutputField,
+    OutputShape,
+    Param,
+    ParamKind,
+    Shape,
+    create_attr_accessor,
+)
+from adaptix._internal.model_tools.introspection.sqlalchemy_tables import get_sqlalchemy_shape
+
+Base = declarative_base()
+
+
+def default_factory():
+    return 2
+
+
+class MyTable(Base):
+    __tablename__ = "MyTable"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    text: Mapped[str]
+    nullable_field: Mapped[Union[int, None]]
+    field_with_default: Mapped[int] = mapped_column(default=2)
+    field_with_default_factory: Mapped[int] = mapped_column(default=default_factory)
+    field_with_default_context_factory: Mapped[int] = mapped_column(default=lambda ctx: 2)
+    field_with_old_syntax = Column(String())
```

Comment:
What abount declarative mapping, mapping via attrs and mapping via dataclass?

Reply:
You mean "imperative", right?